python/lsst/summit/utils/utils.py

```diff
@@ -481,3 +485,118 @@ def getSite():
+    altAz = AltAz(obstime=visitInfo.date.toPython(), location=AUXTEL_LOCATION)
+    obsAltAz = skyLocation.transform_to(altAz)
+    alt = geom.Angle(obsAltAz.alt.degree, geom.degrees)
+    az = geom.Angle(obsAltAz.az.degree, geom.degrees)
+
+    return alt, az
+
+
+def getExpPositionOffset(exp1, exp2, useWcs=True):
+    """Get the change in sky position between two exposures.
+
+    Given two exposures, calculate the offset on the sky between the images.
+    If useWcs then use the (fitted or unfitted) skyOrigin from their WCSs, and
+    calculate the alt/az from the observation times, otherwise use the nominal
+    values in the exposures' visitInfos. Note that if using the visitInfo
+    values that for a given pointing the ra/dec will be ~identical, regardless
+    of whether astrometric fitting has been performed.
+
+    Values are given as exp1-exp2, are directional, and *not* given mod 360
+    so large values should be expected for small offsets depending on the
+    direction.
+
+    Parameters
+    ----------
+    exp1 : `lsst.afw.image.Exposure`
+        The first exposure.
+    exp2 : `lsst.afw.image.Exposure`
+        The second exposure.
+    useWcs : `bool`
+        Use the WCS for the ra/dec and alt/az if True, else use the nominal/
+        boresight values from the exposures' visitInfos.
+
+    Returns
+    -------
+    offsets : `lsst.pipe.base.Struct`
+        A struct contain the offsets:
+            - deltaRa
+            - deltaDec
+            - deltaAlt
+            - deltaAz
+            - deltaPixels
+
+    Notes
+    -----
+    Care has *not* been taken to deal with things mod 360, so beware of large
+    offsets being reported for small results.
+    """
+
+    wcs1 = exp1.getWcs()
+    wcs2 = exp2.getWcs()
+    pixScaleArcSec = wcs1.getPixelScale().asArcseconds()
+    assert np.isclose(pixScaleArcSec, wcs2.getPixelScale().asArcseconds()), \
+           "Pixel scales in the exposures differ."
+
+    if useWcs:
+        p1 = wcs1.getSkyOrigin()
+        p2 = wcs2.getSkyOrigin()
+        alt1, az1 = getAltAzFromSkyPosition(p1, exp1.getInfo().getVisitInfo())
+        alt2, az2 = getAltAzFromSkyPosition(p2, exp2.getInfo().getVisitInfo())
+        ra1 = p1[0]
+        ra2 = p2[0]
+        dec1 = p1[1]
+        dec2 = p2[1]
+    else:
+        az1 = exp1.visitInfo.boresightAzAlt[0]
+        az2 = exp2.visitInfo.boresightAzAlt[0]
+        alt1 = exp1.visitInfo.boresightAzAlt[1]
+        alt2 = exp2.visitInfo.boresightAzAlt[1]
+
+        ra1 = exp1.visitInfo.boresightRaDec[0]
+        ra2 = exp2.visitInfo.boresightRaDec[0]
+        dec1 = exp1.visitInfo.boresightRaDec[1]
+        dec2 = exp2.visitInfo.boresightRaDec[1]
+
+        p1 = exp1.visitInfo.boresightRaDec
+        p2 = exp2.visitInfo.boresightRaDec
+
+    angular_offset = p1.separation(p2).asArcseconds()
+    deltaPixels = angular_offset / pixScaleArcSec
+
+    ret = pipeBase.Struct(deltaRa=ra1-ra2,
```

Comment:
This wrapping needs to be handled or documented. Do you want 23 - 1 == 22 or -2? Should you use `.wrapNear(Angle(0.0))` on the result? (I haven't really checked). Do you handle this downstream?

Reply:
So I tried doing `exp1 - exp2` and `exp2 - exp1` for two visits that are very closeby, and the results looked as I hoped. I then added `.wrapNear()` and the output is unchanged. The tests pass for both. Given that it seems to work both ways and is probably safer with it in, I'll leave it with that addition, and thanks for the tip, I didn't know about `.wrapNear()` 🙂